prometheus/src/main/scala/filodb/prometheus/ast/Expressions.scala

```diff
@@ -28,57 +28,81 @@ trait Expressions extends Aggregates with Functions {
-        throw new UnsupportedOperationException("Binary operations on scalars is not supported yet")
-      }
-
-      (lhs, rhs) match {
-        case (lh: Function, rh: Function) if lh.isScalarFunction() && rh.isScalarFunction() =>
-          val scalar = lh.toSeriesPlan(timeParams).asInstanceOf[ScalarPlan]
-          val seriesPlanRhs = rh.toSeriesPlan(timeParams)
-          ScalarVectorBinaryOperation(operator.getPlanOperator, scalar, seriesPlanRhs, true)
-
-        case (lh: ScalarExpression, rh: PeriodicSeries) =>
-          val scalar = ScalarFixedDoublePlan(lh.toScalar,
-            RangeParams(timeParams.start, timeParams.step, timeParams.end))
-          val seriesPlan = rh.toSeriesPlan(timeParams)
-          ScalarVectorBinaryOperation(operator.getPlanOperator, scalar, seriesPlan, scalarIsLhs = true)
-
-        case (lh: PeriodicSeries, rh: ScalarExpression) =>
-          val scalar = ScalarFixedDoublePlan(rh.toScalar, RangeParams(timeParams.start, timeParams.step,
-            timeParams.end))
-          val seriesPlan = lh.toSeriesPlan(timeParams)
-          ScalarVectorBinaryOperation(operator.getPlanOperator, scalar, seriesPlan, scalarIsLhs = false)
-
-        case (lh: Function, rh: PeriodicSeries) if lh.isScalarFunction() =>
-          val scalar = lh.toSeriesPlan(timeParams).asInstanceOf[ScalarPlan]
-          val seriesPlanRhs = rh.toSeriesPlan(timeParams)
-          ScalarVectorBinaryOperation(operator.getPlanOperator, scalar, seriesPlanRhs, scalarIsLhs = true)
-
-        case (lh: PeriodicSeries, rh: Function) if rh.isScalarFunction =>
-          val scalar = rh.toSeriesPlan(timeParams).asInstanceOf[ScalarPlan]
-          val seriesPlanlhs = lh.toSeriesPlan(timeParams)
-          ScalarVectorBinaryOperation(operator.getPlanOperator, scalar, seriesPlanlhs, scalarIsLhs = false)
-
-        case (lh: PeriodicSeries, rh: PeriodicSeries) =>
-          val seriesPlanLhs = lh.toSeriesPlan(timeParams)
-          val seriesPlanRhs = rh.toSeriesPlan(timeParams)
-          val cardinality = if (operator.getPlanOperator.isInstanceOf[SetOperator])
-            Cardinality.ManyToMany
-          else
-            vectorMatch.map(_.cardinality.cardinality).getOrElse(Cardinality.OneToOne)
-
-          val matcher = vectorMatch.flatMap(_.matching)
-          val onLabels = matcher.filter(_.isInstanceOf[On]).map(_.labels)
-          val ignoringLabels = matcher.filter(_.isInstanceOf[Ignoring]).map(_.labels)
-
-          BinaryJoin(seriesPlanLhs, operator.getPlanOperator, cardinality, seriesPlanRhs,
-            onLabels.getOrElse(Nil), ignoringLabels.getOrElse(Nil),
-            vectorMatch.flatMap(_.grouping).map(_.labels).getOrElse(Nil))
-
-        case _ => throw new UnsupportedOperationException("Invalid operands")
+      if (hasScalarResult(lhs) && hasScalarResult(rhs)) {
+        val rangeParams = RangeParams(timeParams.start, timeParams.step, timeParams.end)
+
+        (lhs, rhs) match {
```

Comment:
How is
`case (lh: ScalarExpression, rh: BinaryExpression)`

covered ?

Reply:
Ohh yes. I missed this. Added along with test case. Thanks!